### PR TITLE
l2geth: delete old genesis setter

### PR DESCRIPTION
**Pull request by tynes**
**Description**
Removes crufty code that allows the genesis setter to be backwards compatible with the old `OVM_ETH`. Since we are now past that `OVM_ETH` being used, we no longer need to maintain backwards compatibility.

### CODE REVIEW

**Review by snario — APPROVED** [review]

File: .changeset/rotten-peaches-dance.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@eth-optimism/l2geth': patch
```
Comment (karlfloersch) [inline .changeset/rotten-peaches-dance.md:2]: It pains me that this is considered a patch but I presume that it is only when we reach 1.0 that we will move away from patches being backwards incompatible?

**Review by karlfloersch — APPROVED** [review]
This LGTM.

Added a comment on how this breaks backwards compatibility but is considered a "patch" but that is fine it seems for now. Or at least it requires a bigger discussion as I have noticed this in a number of PRs.

File: l2geth/core/genesis.go [hunk 2]
```diff
@@ -302,19 +302,11 @@ func ApplyOvmStateToState(statedb *state.StateDB, stateDump *dump.OvmDump, l1XDo
...
-			log.Info("Detected legacy OVM_ETH dump, setting slot 0x8")
-			l1GatewaySlot := common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000000008")
-			l1GatewayValue := common.BytesToHash(l1ETHGatewayAddress.Bytes())
-			statedb.SetState(OVM_ETH.Address, l1GatewaySlot, l1GatewayValue)
-		}
+		// Set the gateway of OVM_ETH at new dump
+		log.Info("Detected current OVM_ETH dump, setting slot 0x1 ")
+		l1GatewaySlot := common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000000001")
+		l1GatewayValue := common.BytesToHash(l1ETHGatewayAddress.Bytes())
+		statedb.SetState(OVM_ETH.Address, l1GatewaySlot, l1GatewayValue)
```
Comment (karlfloersch) [inline l2geth/core/genesis.go:309]: Worth noting that these 'gateway' variables should likely be renamed to 'bridge' at some point in the future.

File: .changeset/rotten-peaches-dance.md (hunk 1 above)
Comment (snario) [reply .changeset/rotten-peaches-dance.md:2]: It's `minor`

**Comment by tynes** [comment]
This PR can be closed due to being superseded by another PR

**Pull request closed without merging**